api/operator/v1beta1/vmsingle_types.go

```diff
@@ -56,10 +56,11 @@ type VMSingleSpec struct {
 	// +optional
 	RemovePvcAfterDelete bool `json:"removePvcAfterDelete,omitempty"`
 
-	// RetentionPeriod for the stored metrics
-	// Note VictoriaMetrics has data/ and indexdb/ folders
-	// metrics from data/ removed eventually as soon as partition leaves retention period
-	// reverse index data at indexdb rotates once at the half of configured [retention period](https://docs.victoriametrics.com/Single-server-VictoriaMetrics/#retention)
+	// RetentionPeriod defines how long to retain stored metrics, specified as a duration (e.g., "1d", "1w", "1m").
+	// Data with timestamps outside the RetentionPeriod is automatically deleted. The minimum allowed value is 1d, or 24h.
+	// The default value is 1 (one month).
+	// See [retention](https://docs.victoriametrics.com/victoriametrics/single-server-victoriametrics/#retention) docs for details.
+	// +optional
```

Comment:
I think, it's worth to add validation pattern:

`// +kubebuilder:validation:Pattern:="[0-9]+(h|d|y)"`
https://book.kubebuilder.io/reference/generating-crd.html?highlight=validation#validation

 and change json tag for ``json:"retentionPeriod,omitempty"`. Previously, `omitempty` tag was used as an optional indicator for field.

Reply:
@f41gh7 added this regexp `[0-9]+(h|d|y)*` (see * in the end) because retentionPeriod accept just a number (a month unit is assumed in this case)

Reply:
WDYT of changing regex to:

```
^[0-9]+(h|d|y)?$
```
To should cover possible cases with optional characters and also forbid any other symbols.